### PR TITLE
refactor(config): split zk-related configuration

### PR DESCRIPTION
**Pull request by Karrq**
## Motivation
All zksync related configurations are flattened into the top level `Config` struct, having to deal with duplicated names and need to keep relatively simplified values to encode wanted behavior.

## Solution
Encapsulate zksync related configuration as to not pollute the top level `Config` struct, allowing a more structured approach to configuration.

This is how `foundry.toml` _can_ look like, with the changes:
```
[zksync] #all configuration below lives inside the `zksync` item
enable = false # can still be enabled from the cli
zksolc = "1.5.0"
force_evmla = true
fallback_oz = true
solc_path = "./solc-0.8.23-1.0.1"
optimizer = true
optimizer_mode = 'z'
```
before:
```
# no section
zksync = true
zksolc = "1.5.0"
force_evmla = true
fallback_oz = true
zk_solc_path = "./solc-0.8.23-1.0.1"
zk_optimizer = true
mode = 'z'
```

### CODE REVIEW

**Comment by Karrq** [comment]
Unfortunately there's a bit of attrition on converting the cli args to the config values, maybe we can find a way to improve that so future changes to zksync configuration can be further simplified or at least be done in the same context

**Comment by elfedy** [comment]
My two cents: Having everything under `[zksync]` might be a good idea as it provides a clean separation for the config. The further nesting might be a bit more controversial because:

1. It's hard to get the "right" level of nesting on the first try, more so when we are in an alpha stage and options come and go all the time, we would  have to change categories a bit and might provide more confusion than clarity in the end.
2. We don't have THAT many zksync specific options, a single level might even be cleaner
3. (This is the most important to me) Foundry's API does not use this nesting and we should probably aim to mirror Foundry's style at least in spirit for a more cohesive experience

**Comment by Karrq** [comment]
Okay, I can refactor to flatten the structure so we keep everything under one `[zksync]` section

File: crates/cli/src/opts/build/core.rs [hunk 1]
```diff
@@ -135,6 +135,38 @@ impl CoreBuildArgs {
...
+
+        macro_rules! set_if_some {
+            ($src:expr, $dst:expr) => {
+                if let Some(src) = $src {
+                    $dst = src.into();
+                }
+            };
+        }
+
+        set_if_some!(compiler_cfg.zksync.then_some(true), zksync.enable);
```
Comment (elfedy) [inline crates/cli/src/opts/build/core.rs:151]: If we are keeping the cli maybe prefix everything with `zksync` and the same name as the config.

Comment (elfedy) [reply crates/cli/src/opts/build/core.rs:151]: We could also move zksync specific stuff to another struct and do the figment magic there instead?

Comment (Karrq) [reply crates/cli/src/opts/build/core.rs:151]: yeah and `#[serde(flatten)]` it too, nice

**Review by elfedy — APPROVED** [review]

File: crates/cli/src/opts/build/core.rs (hunk 1 above)
Comment (elfedy) [reply crates/cli/src/opts/build/core.rs]: Nice, maybe also rename `zksync` toml item to `zk` just so everything is named the same.

**Comment by elfedy** [comment]
If we merge this we should probably notify somewhere about the API change

File: crates/cli/src/opts/build/core.rs (hunk 1 above)
Comment (Karrq) [reply crates/cli/src/opts/build/core.rs]: I would have put `zksync` in the cli also, but some keys are already long enough, but in the config it makes it very easy to see this is _zksync_ specific configuration

**Pull request merged**